test/testdrive/unittest_source.td

```diff
@@ -0,0 +1,31 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# This testdrive file uses the deprecated syntax, but is otherwise identical to upsert-kafka-new.td
+#
+# This file can be deleted when/if we finish the deprecation and perform the removal of the old syntax.
+> CREATE CONNECTION c_conn
+  FOR CONFLUENT SCHEMA REGISTRY URL '${testdrive.schema-registry-url}'
+
+> CREATE SOURCE unit
+  FROM UNITTEST
+  '[
+    {"key": "fish", "val": "value", "offset": 0},
+    {"key": "fish2", "val": "hmm", "offset": 1},
+    {"key": "fish", "val": "value2", "offset": 2, "pend_forever": true}
+  ]'
```

Comment:
Can you add documentation somewhere about the syntax + semantics of the test script? Also, I recommend going with a more command-based approach:

```rust
#[serde(tag = "command")]
#[serde(rename_all = "lowercase")]
enum ScriptCommand {
    Emit { key: String, value: String, offset: ... },
    Block,
}

{ "command": "emit",  "key": "fish", "value": "value", "offset": 0}
{ "command": "block" }
```